### PR TITLE
cloud: Get rid of the joinMachine struct

### PR DESCRIPTION
**Pull request by ejj**
It's not adding a lot of value over db.Machine.

### CODE REVIEW

File: cloud/cloud.go [hunk 1]
```diff
@@ -153,18 +153,18 @@ func (cld cloud) runOnce() {
 }
 
 func (cld cloud) boot(machines []db.Machine) {
-	var cloudMachines []joinMachine
+	// As a defensive measure, make sure set everything to zero that a provider
```
Comment (kklin) [inline cloud/cloud.go]: I'm not sure what this comment means

Comment (ejj) [reply cloud/cloud.go]: Updated the comment, make sense now?

Comment (kklin) [reply cloud/cloud.go]: Kind of -- I don't get what we're defending against

Comment (ejj) [reply cloud/cloud.go]: We're just defending against provider writers from depending on machine fields that they shouldn't. I.E. breaking layering boundaries which can lead to icky spaghetti code.  This is why machine.Machine existed in the first place -- it's generally not best to give a module information it shouldn't be aware of.

**Pull request merged**